app/v2/notifications/post_notifications.py

```diff
@@ -331,7 +331,10 @@ def process_letter_notification(
     get_pdf_for_templated_letter.apply_async([str(notification.id)], queue=queue)
 
     if test_key and not current_app.config["SEND_LETTERS_ENABLED"]:
-        create_fake_letter_response_file.apply_async((notification.reference,), queue=queue)
+        create_fake_letter_callback.apply_async(
```

Comment:
I'm not quite sure I understand why using the new task needs to come in the second PR. With things as they currently are in this PR, the new task gets created and used immediately, but the old task definition still exists, so will deal with any of the old tasks still in existence.

If using the new task moves to the second PR, then nothing changes about how the code works here. We would start using the new task at the same time that the old one is deleted (after PR 2 is deployed) which seems like it could lead to the situation we are trying to avoid - old tasks existing and having nothing to process them.

Reply:
i think the issue i was thinking about is this (i'm not 100% from memory if this is the case, but i'm pretty sure)

1. deploy starts
2. new api-web handles a test letter, and puts a new `create-fake-letter-callback` task on the `research-mode-tasks` queue
3. old api-worker-research reads from `research-mode-tasks` and receives a new `create-fake-letter-callback` task
4. old api-worker-research raises an exception, because it doesn't have a function definition to execute for that task
5. task is lost(?)

Reply:
Right, thanks - I see what you mean now. I was thinking about what would happen to the old tasks during deployment, but had forgotten that the new tasks might have issues too. I've updated this to stop calling the new task